### DISCUSSION TITLE
When will arm64-based ubuntu be supported?

### DISCUSSION THREAD

**Discussion by mengchaoheng** [cyberbotics/webots#4416] [Q&A]
I noticed that Webots is provided for Linux 64 (x86-64) systems. But I would love to support arm64.

**Comment by omichel** [comment]
We plan to provide a arm64 version of Webots for macOS, to be released in June 2022.
However, we have no plan to provide a arm64 version of Webots for Linux.
But you might be able to create it by yourself from the source code of Webots.

**Reply by mengchaoheng** [reply]
I don't understand why it is more urgent to provide macOS support than arm-based ubuntu, are there more m1 mac users than arm-based ubuntu users? @omichel

**Reply by omichel** [reply]
Yes, I think so.

**Reply by mengchaoheng** [reply]
> Yes, I think so.

But all developers who use m1 mac will install the ubuntu virtual machine for development, such as developing ros applications, I hope the support of arm-based ubuntu can be realized as soon as possible.

**Reply by omichel** [reply]
Webots is fully open-source and you are welcome to contribute the port to arm-based Ubuntu. We will be happy to review any PR you may contribute.

**Reply by zegangYang** [reply]
I am trying build webots on my mac studio（m1 max），and there still have some lib need to be build manaully,and i am willing to provide some docs or tips later

**Comment by JACOBIN-SCTCS** [comment]
If somebody is working on macOS and developing ROS applications,  you can use Docker. Follow the instructions in https://github.com/cyberbotics/webots_ros2/wiki/Complete-Installation-Guide.

**Comment by romain130492** [comment]
any news about this? We now have m1-m2-m3 chips;)

**Reply by CoolSpy3** [reply]
Arm-based mac support was introduced in #4543 and released in R2022b. There has not been any progress towards an arm-based Ubuntu release.

**Comment by myudak** [comment]
bruhhh i need webots arm64
